core/src/main/java/org/testcontainers/containers/Network.java

```diff
@@ -0,0 +1,107 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.command.CreateNetworkCmd;
+import lombok.*;
+import org.junit.rules.ExternalResource;
+import org.junit.rules.TestRule;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.utility.ResourceReaper;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+public interface Network extends AutoCloseable, TestRule {
+
+    String getId();
+
+    String getName();
+
+    Boolean getEnableIpv6();
+
+    String getDriver();
+
+    boolean isCreated();
+
+    default boolean create() {
+        return getId() != null;
```

Comment:
`id` is the only thing needed to use this network. Default implementation assumes "no creation needed" semantic and returns true if id exists. **Do not** read the implementation to understand the idea of default implementation here.

Example:
```
class ExistingNetwork implements Network {
    String getId() {
        return "some-predefined-id";
    }
}
```